src/main/java/Steps/BasePage.java

```diff
@@ -16,4 +24,20 @@ default String getCurrentUrl() {
     default void setMaxWindowResolution() {
         BrowserСhoice.getDriver().manage().window().maximize();
     }
+
+    default <T> T waitUntilProductPageWillBeLoading() {
+        WebElement product = BrowserСhoice.getDriver()
+                .findElement(By.xpath("//div[contains(@class,'full-width df jcc aic slide')]"));
+        Wait<WebDriver> wait = new WebDriverWait(BrowserСhoice.getDriver(), Duration.ofSeconds(4));
+        wait.until(ExpectedConditions.visibilityOf(product));
```

Comment:
It won't work like that because first line in this method contains `findElement` call which will cause driver to search for that element and throw exception if it wasn't found in ~0.5 sec. So it will not reach to waiter code.